expr/parse_test.go

```diff
@@ -325,6 +325,22 @@ func TestParse(t *testing.T) {
 			},
 			nil,
 		},
+		{
+			"func(metric, key2='true', key1='false')",
+			&expr{
+				str:   "func",
+				etype: etFunc,
+				args: []*expr{
+					{str: "metric"},
+				},
+				namedArgs: map[string]*expr{
+					"key2": {etype: etString, str: "true"},
+					"key1": {etype: etString, str: "false"},
```

Comment:
don't we want these to be etBool ?

Reply:
At parse time, we don't know what type we want. We only know what type they are. Only when they are converted to expressions do we know what signature to apply and how to match them up.